bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,427 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cbus.handler;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+import org.openhab.binding.cbus.internal.CBusCGateConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daveoxley.cbus.CGateConnectException;
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.CGateInterface;
+import com.daveoxley.cbus.CGateSession;
+import com.daveoxley.cbus.CGateThreadPool;
+import com.daveoxley.cbus.CGateThreadPoolExecutor;
+import com.daveoxley.cbus.events.EventCallback;
+import com.daveoxley.cbus.status.StatusChangeCallback;
+
+/**
+ * The {@link CBusThreadPool} is responsible for executing jobs from a threadpool
+ *
+ * @author John Harvey - Initial contribution
+ */
+
+@NonNullByDefault
+final class CBusThreadPool extends CGateThreadPool {
+
+    private final Map<@Nullable String, @Nullable CGateThreadPoolExecutor> executorMap = new HashMap<>();
+
+    @Override
+    protected synchronized CGateThreadPoolExecutor CreateExecutor(@Nullable String name) {
+        @Nullable
+        CGateThreadPoolExecutor executor = executorMap.get(name);
+
+        if (executor != null) {
+            return executor;
+        }
+        CGateThreadPoolExecutor newExecutor = new CBusThreadPoolExecutor(name);
+        executorMap.put(name, newExecutor);
+        return newExecutor;
+    }
+
+    @NonNullByDefault
+    public class CBusThreadPoolExecutor extends CGateThreadPoolExecutor {
+        private final ThreadPoolExecutor threadPool;
+
+        public CBusThreadPoolExecutor(@Nullable String poolName) {
```

Comment:
Only because the CGateThreadPoolExecutor does not add the annotations. So if i remove the @Nullable it doesnt compile. 
`/home/jpharvey/openhab/cbus_fix/bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java:[63,2291] Illegal redefinition of parameter name, inherited method from CGateThreadPool does not constrain this parameter`

I am open to suggestions about how to fix it but didnt want to add the Null annotations to the cgate interface class.
I've added a check for null or empty names and made it use a default name in that case. So now the keys cant be null.